src/java.base/share/classes/java/lang/StringUTF16.java

```diff
@@ -412,12 +412,11 @@ public static int compareToCI_Latin1(byte[] value, byte[] other) {
     }
 
     public static int hashCode(byte[] value) {
-        int h = 0;
-        int length = value.length >> 1;
-        for (int i = 0; i < length; i++) {
-            h = 31 * h + getChar(value, i);
+        if (value.length == 0) {
+            return 0;
+        } else {
+            return ArraysSupport.vectorizedHashCode(value, ArraysSupport.UTF16);
```

Comment:
Special case for 1 missing here.

Reply:
Intentionally left out. Array length is always even for `UTF16` arrays, but we could add a case for `2` that'd return `getChar(bytes, 0)` but I didn't see much of a win when I tested this.

Reply:
I do see a 1.5x gain with this special case added:
       return switch (value.length) {
            case 0 -> 0;
            case 2 -> getChar(value, 0);
            default -> ArraysSupport.vectorizedHashCode(value, ArraysSupport.UTF16);
        };
before: 0.987 ns/op 
after: 0.640 ns/op